var/spack/repos/builtin/packages/xsdktrilinos/package.py

```diff
@@ -0,0 +1,121 @@
+    version('12.8.1', '9cc338ded17d1e10ea6c0dc18b22dcd4')
+    version('12.6.4', '44c4c54ccbac73bb8939f68797b9454a')
+
+    def url_for_version(self, version):
+        return '%s/trilinos-release-%s.tar.gz' % \
+            (Xsdktrilinos.base_url, version.dashed)
+
+    variant('hypre',        default=True,
+            description='Compile with Hypre preconditioner')
+    variant('petsc',        default=True,
+            description='Compile with PETSc solvers')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+    variant('debug',        default=False,
+            description='Builds a debug version of the libraries')
+
+    depends_on('cmake', type='build')
+
+    # MPI related dependencies
+    depends_on('mpi')
+    depends_on('hypre', when='+hypre')
+    depends_on('petsc', when='+petsc')
+    depends_on('trilinos')
+
+    def install(self, spec, prefix):
+        options = []
+        options.extend(std_cmake_args)
+
+        mpi_bin = spec['mpi'].prefix.bin
+        # Note: -DXYZ_LIBRARY_NAMES= needs semicolon separated list of names
+        options.extend([
+            '-DxSDKTrilinos_VERBOSE_CONFIGURE:BOOL=OFF',
+            '-DxSDKTrilinos_ENABLE_TESTS:BOOL=ON',
+            '-DxSDKTrilinos_ENABLE_EXAMPLES:BOOL=ON',
+            '-DTrilinos_INSTALL_DIR=%s' % spec['trilinos'].prefix,
+            '-DCMAKE_BUILD_TYPE:STRING=%s' % (
+                'DEBUG' if '+debug' in spec else 'RELEASE'),
+            '-DBUILD_SHARED_LIBS:BOOL=%s' % (
+                'ON' if '+shared' in spec else 'OFF'),
+            '-DTPL_ENABLE_MPI:BOOL=ON',
+            '-DMPI_BASE_DIR:PATH=%s' % spec['mpi'].prefix,
+            '-DxSDKTrilinos_ENABLE_CXX11:BOOL=ON',
+            '-DTPL_ENABLE_HYPRE:BOOL=%s' % (
+                'ON' if '+hypre' in spec else 'OFF'),
+            '-DTPL_ENABLE_PETSC:BOOL=%s' % (
+                'ON' if '+petsc' in spec else 'OFF')
+        ])
+
+        # Fortran lib
+        if spec.satisfies('%gcc') or spec.satisfies('%clang'):
+            libgfortran = os.path.dirname(os.popen(
+                '%s --print-file-name libgfortran.a' %
+                join_path(mpi_bin, 'mpif90')).read())
+            options.extend([
+                '-DxSDKTrilinos_EXTRA_LINK_FLAGS:STRING=-L%s/ -lgfortran' % (
+                    libgfortran),
+                '-DxSDKTrilinos_ENABLE_Fortran=ON'
+            ])
+
+        # for build-debug only:
+        # options.extend([
+        #    '-DCMAKE_VERBOSE_MAKEFILE:BOOL=TRUE'
+        # ])
+
+        with working_dir('spack-build', create=True):
+            cmake('..', *options)
+            make()
+            make('install')
+
+            # The shared libraries are not installed correctly on Darwin;
+            # correct this
+            if (sys.platform == 'darwin') and ('+shared' in spec):
+                fix_darwin_install_name(prefix.lib)
```

Comment:
There is something you can put in the Cmake file to automatically do the darwin fix business so you shouldn't need it here.